src/main/java/org/sonarsource/sonarlint/ls/standalone/notifications/PromotionalNotifications.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * SonarLint Language Server
+ * Copyright (C) 2009-2024 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonarsource.sonarlint.ls.standalone.notifications;
+
+import java.util.List;
+import java.util.Locale;
+import java.util.Objects;
+import org.eclipse.lsp4j.DidOpenTextDocumentParams;
+import org.sonarsource.sonarlint.core.commons.Language;
+import org.sonarsource.sonarlint.ls.EnginesFactory;
+import org.sonarsource.sonarlint.ls.SonarLintExtendedLanguageClient;
+import org.sonarsource.sonarlint.ls.settings.SettingsManager;
+
+public class PromotionalNotifications {
+  private final SonarLintExtendedLanguageClient client;
+  private final SettingsManager settingsManager;
+
+  public PromotionalNotifications(SonarLintExtendedLanguageClient client, SettingsManager settingsManager) {
+    this.client = client;
+    this.settingsManager = settingsManager;
+  }
+
+  public void didOpen(DidOpenTextDocumentParams didOpenTextDocumentParams) {
+    var isConnected = settingsManager.hasConnectionDefined();
+    if (!isConnected) {
+      var connectedLanguageForOpenedFile = EnginesFactory.getConnectedLanguages().stream().filter(additionalLanguage ->
+        Objects.equals(additionalLanguage.getLanguageKey().toLowerCase(Locale.ENGLISH),
+          didOpenTextDocumentParams.getTextDocument().getLanguageId().toLowerCase(Locale.ENGLISH))).findFirst();
```

Comment:
I believe this (c|s)hould be extracted in a method.

First, the language ID sent by the client should be mapped to the Sonar language - see `AnalysisClientInputFile#toSqLanguage` (and maybe this could be extracted somewhere else too).

Second, the "filtering on connected languages" logic could be encapsulated in a (static) method from `EnginesFactory`, e.g. `isAvailableInConnectedMode(Language)`.